volatility3/framework/automagic/module.py

```diff
@@ -62,5 +62,17 @@ def __call__(
                 offset = context.config[layer_kvo_config_path]
                 context.config[offset_config_path] = offset
 
+                # macOS KernelCache module override
+                layer_class_config_path = interfaces.configuration.path_join(
+                    new_config_path, req, "class"
+                )
+                layer_class_str = context.config[layer_class_config_path]
+                if (
+                    layer_class_str
+                    == "volatility3.framework.layers.intel.MacIntelMhFilesetKernelCache"
```

Comment:
I was trying to do a normal class/Type comparison, when I realized everything in the config was indeed a string. So I decided to mimic : 

https://github.com/volatilityfoundation/volatility3/blob/55dd39f2ba60ffdd2126b7ea011940f0df42815a/volatility3/framework/automagic/module.py#L37-L39

I am currently checking for methods of converting an object to a full string path, which might be cleaner than the previous snippet of code :

`"."join(contexts.Module.__module__, contexts.Module.__qualname__)`. 

But converting `context.config[layer_class_config_path]` string to a Python class might involve dynamic string to object manipulation, which is neither really great ?

Reply:
Yeah, it should end up as a string in the config, but we want to make sure that string relates to an existing class and that if the file gets moved/renamed etc, an IDE doing refactor can spot the break, hence converting it from an object to a string name.  So, yeah, basically convert the python object to the string name (we can make a separate function for clarity) but then compare `layer_class_str == class_string_name(class_we_care_about)` rather than a fixed string.

You shouldn't ever need to convert from the string to a class?

Reply:
I see, I will change this bit, but after your update, as we might remove the `MacIntelMhFilesetKernelCache` and use something that makes more sense.